my-app/src/app/app.component.spec.ts

```diff
@@ -1,31 +1,82 @@
-import { TestBed } from '@angular/core/testing';
-import { AppComponent } from './app.component';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
 
-describe('AppComponent', () => {
-  beforeEach(async () => {
+import {FormsModule} from "@angular/forms";
+import {BrowserModule} from "@angular/platform-browser";
+
+import isArray from "lodash/isArray";
+
+import {AppComponent} from './app.component';
+
+import {HeaderComponent} from "./components/header/header.component";
+import {FooterComponent} from "./components/footer/footer.component";
+import {LoadMoreButtonComponent} from "./components/load-more-button/load-more-button.component";
+import {SectionComponent} from "./components/section/section.component";
+import {CoursesComponent} from "./components/courses/courses.component";
+import {BreadcrumbsComponent} from "./components/breadcrumbs/breadcrumbs.component";
+import {Course, DurationHandlerPipe} from "./shared";
+
+import Spy = jasmine.Spy;
+
+describe('AppComponent', (): void => {
+  let app: AppComponent;
+  let fixture: ComponentFixture<AppComponent>;
+
+  beforeEach(async (): Promise<void> => {
     await TestBed.configureTestingModule({
       declarations: [
-        AppComponent
+        AppComponent,
+        HeaderComponent,
```

Comment:
You add all components that are nested in the app component. So it's not a unit test anymore, but integration. I would suggest using NO_ERRORS_SCHEMA to just ignore unknown elements in the template.
Imagine that some of these components will become dependent on some services. Then to make this test pass you'll have to mock these services here - it's not convenient. Good unit test should be isolated.